backend-rust/src/rest_api.rs

```diff
@@ -0,0 +1,144 @@
+use std::sync::Arc;
+
+use crate::graphql_api::AccountStatementEntryType;
+use axum::{
+    extract::{Query, State},
+    http::HeaderName,
+    response::{AppendHeaders, IntoResponse},
+    routing::get,
+    Router,
+};
+use chrono::{DateTime, TimeDelta, Utc};
+use concordium_rust_sdk::{common::types::Amount, id::types::AccountAddress};
+use futures::TryStreamExt as _;
+use reqwest::StatusCode;
+use sqlx::PgPool;
+use tower_http::cors::{Any, CorsLayer};
+
+pub struct Service {
+    pool: PgPool,
+}
+
+const EXPORT_MAX_DAYS: i64 = 31;
+
+impl Service {
+    pub fn new(pool: PgPool) -> Self {
+        Self {
+            pool,
+        }
+    }
+
+    pub fn as_router(self) -> Router {
+        let cors_layer = CorsLayer::new()
+            .allow_origin(Any)  // Open access to selected route
+            .allow_methods(Any)
+            .allow_headers(Any);
+        Router::new()
+            .route("/rest/export/statement", get(Self::export_account_statements))
+            .layer(cors_layer)
+            .with_state(self.pool)
+    }
+
+    async fn export_account_statements(
+        Query(params): Query<ExportAccountStatement>,
+        State(pool): State<PgPool>,
+    ) -> ApiResult<(AppendHeaders<[(HeaderName, String); 2]>, String)> {
+        let to = params.to_time.unwrap_or_else(|| chrono::Utc::now());
+        let from = params.from_time.unwrap_or_else(|| to - TimeDelta::days(EXPORT_MAX_DAYS));
+        if to - from > TimeDelta::days(EXPORT_MAX_DAYS) {
+            return Err(ApiError::ExceedsMaxAllowed);
+        }
+        let mut rows = sqlx::query_as!(
+            ExportAccountStatementEntry,
+            r#"SELECT
+                blocks.slot_time as timestamp,
+                account_statements.amount,
+                account_statements.account_balance,
+                entry_type as "entry_type: AccountStatementEntryType"
+            FROM accounts
+                JOIN account_statements
+                    ON accounts.index = account_statements.account_index
+                JOIN blocks
+                    ON blocks.height = account_statements.block_height
+            WHERE
+                accounts.address = $1
+                AND slot_time > $2
```

Comment:
I think we need to include the boundary in a bucket.
```suggestion
                AND slot_time >= $2
```

Reply:
Yes, that makes sense.
.NET actually includes both, so I'll do the same.